From: Sakura286 <sakura286@outlook.com>
Date: Thu, 12 Jun 2025 17:18:29 +0800
Subject: enable offload-compress

---
 CMakeLists.txt         | 1 +
 library/CMakeLists.txt | 5 +++++
 2 files changed, 6 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 9cbf85e..1352d1a 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -77,6 +77,7 @@ option(BUILD_MEMSTAT "Build rocSPARSE with memory statistics enabled" OFF)
 option(BUILD_ROCSPARSE_ILP64 "Build rocSPARSE with rocsparse_int equal to int64_t" OFF)
 option(BUILD_COMPRESSED_DBG "Enable compressed debug symbols" ON)
 option(BUILD_WITH_ROCBLAS "Enable building rocSPARSE with rocBLAS" ON)
+option(BUILD_OFFLOAD_COMPRESS "Build with offload compress" OFF)
 
 # Clients utilize rocsparse fortran API and a fortran compiler
 if( NOT BUILD_FORTRAN_CLIENTS )
diff --git a/library/CMakeLists.txt b/library/CMakeLists.txt
index 9ddc654..cb58d90 100644
--- a/library/CMakeLists.txt
+++ b/library/CMakeLists.txt
@@ -73,6 +73,11 @@ add_library(roc::rocsparse ALIAS rocsparse)
 # Target compile options
 target_compile_options(rocsparse PRIVATE -Wno-unused-command-line-argument -Wall)
 
+if ( BUILD_OFFLOAD_COMPRESS )
+  target_compile_options( rocsparse PRIVATE --offload-compress )
+endif ()
+
+
 # Target include directories
 target_include_directories(rocsparse
                            PRIVATE $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/src/include>
